### PR TITLE
HARMONY-1363: Add granuleName parameter support

### PR DESCRIPTION
**Pull request by indiejames**
## Jira Issue ID
HARMONY-1363

## Description
Add support for the `granuleName` parameter

## Local Test Steps
1. checkout this branch
2. use `pip` to install the local updates to harmony-py
```
cd harmony-py
pip install .
```
3. run the basic.ipynb notebook (only need to run the first few cells to do the imports and the create the client, then can skip to the last four code cells to test the `granuleName` requests).

## PR Acceptance Checklist
* [x] Acceptance criteria met
* [x] Tests added/updated (if needed) and passing
* [x] Documentation updated (if needed)

### CODE REVIEW

**Review by ygliuvt — APPROVED** [review]

File: examples/basic.ipynb [hunk 1]
```diff
@@ -570,7 +656,12 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.8.2"
+   "version": "3.10.9"
+  },
+  "vscode": {
+   "interpreter": {
+    "hash": "341b7d2e50a6ee8d836f143dcf87119dfe72f0053ce895c8752bf7a40b324b52"
```
Comment (vinnyinverso) [inline examples/basic.ipynb:663]: Do we want to leave this here or not?

File: requirements/dev.txt [hunk 2]
```diff
@@ -1,17 +1,16 @@
...
-m2r2 ~= 0.2
-nose ~= 1.3
-PyYAML ~= 5.4
-pytest ~= 6.2
-pytest-cov ~= 2.11
-pytest-mock ~= 3.5
-pytest-watch ~= 4.2
-responses ~= 0.12
+autopep8
+camel-case-switcher
```
Comment (vinnyinverso) [inline requirements/dev.txt:2]: What's the thinking behind removing the versions?

Comment (indiejames) [reply requirements/dev.txt]: don't bother reviewing this yet - I have made several commits trying to resolve snyk issues without breaking everything with no luck. Unfortunately I don't know how to run snyk on the python deps without pushing to github.

File: examples/basic.ipynb (hunk 1 above)
Comment (indiejames) [reply examples/basic.ipynb:663]: no

**Review by vinnyinverso — APPROVED** [review]

**Pull request merged**